androidvalidatedforms/src/main/java/com/azavea/androidvalidatedforms/controllers/ImageController.java

```diff
@@ -0,0 +1,385 @@
+            Cursor cursor = loader.loadInBackground();
+            int col_idx = cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.DATA);
+            cursor.moveToFirst();
+            String imagePath = cursor.getString(col_idx);
+            cursor.close();
+            Log.d(LOG_LABEL, "Have image path: " + imagePath);
+
+            // store image path to model
+            getModel().setValue(getName(), imagePath);
+            setNeedsValidation();
+
+            setImageToPath(imagePath);
+
+        } else {
+            Log.w(LOG_LABEL, "got unrecognized intent result");
+        }
+    }
+
+    /**
+     * Create a file to use for storing a photo taken by the camera.
+     *
+     * @return File created in media directory, in a subdirectory that is the app name
+     * @throws IOException
+     */
+    private File createImageFile() throws IOException {
+        String appName = getContext().getApplicationInfo().loadLabel(getContext().getPackageManager()).toString();
+
+        if (Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)) {
+            File picturePath = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES);
+
+            File mediaStorageDir = new File(picturePath, appName);
+
+            if (!mediaStorageDir.exists()) {
+                if (!mediaStorageDir.mkdirs()) {
+                    Log.e(LOG_LABEL, "failed to create directory");
+                    return null;
+                }
+            }
+
+            // Create a media file name
+            String timeStamp = IMAGE_DATE_FORMAT.format(new Date());
+            File imageFile = File.createTempFile("IMG_" + timeStamp + "_", ".jpg", mediaStorageDir);
+            currentPhotoPath = Uri.fromFile(imageFile);
+
+            return imageFile;
+        } else {
+            Log.e(LOG_LABEL, "No external media mounted or emulated");
+            currentPhotoPath = null;
+            return null;
+        }
+    }
+
+    /**
+     * Helper to downscale an image and put it into a view.
+     * See: http://developer.android.com/training/camera/photobasics.html
+     *
+     * @param view ImageView to set (can also be an ImageButton, which is a subclass of ImageView)
+     * @param imagePath Path to the image to set into the view
+     * @return true on success
+     */
+    public static boolean setDownscaledImageFromFilePath(ImageView view, String imagePath, int minWidth, int minHeight) {
+        // first check if image file actually exists
+        File file = new File(imagePath);
+        if (!file.exists()) {
+            Log.e(LOG_LABEL, "Image file does not exist at " + imagePath);
+            view.setVisibility(View.GONE);
+            return false;
+        }
+
+        // Get the dimensions of the view
+        int targetW = view.getWidth();
+        int targetH = view.getHeight();
+
+        if (targetW == 0) {
```

Comment:
This is to avoid division by zero, if the `ImageView` has no size yet, but I suppose it might as well check for minimum size instead, yes.